### ISSUE TITLE
[FL autosuggest prototype] The location input must announce errors to screenreaders

### ISSUE THREAD

**Issue by qjac** [department-of-veterans-affairs/va.gov-cms#22473]
## Describe the defect
Found as part of [ADE accessibility audit](https://github.com/department-of-veterans-affairs/va.gov-team/issues/120537#issuecomment-3347814120) of the location search box, Facility type select, custom select, and autosuggest fields on the [autosuggest prototype](https://staging.va.gov/find-locations/).

> **Issue: Error message not announced using VoiceOver**
> **Field:** Location search box
> **Expected severity level:** 1
> **Expected experience standard:** User can perceive all meaningful elements
> **Description:** When triggering the onblur error, MacOS VoiceOver doesn’t announce the error message. It also doesn’t announce it when focus is moved to the field after pressing “Search.” [Clip](https://coformaco.zoom.us/clips/share/jCno6zX0Spy8IVMftvHhMw)
> **How to recreate:** Turn on VoiceOver, TAB into and out of the field
> **Recommendation:** [Same recommendation as above](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22458#issue-3497017460) - switch to <va-text-input> if possible, emulate its error handling if not.

[Location input is a custom combox](https://dsva.slack.com/archives/C0FQSS30V/p1759523788832689?thread_ts=1759522358.548369&cid=C0FQSS30V) and is therefore not inheriting the error handling from VADS inputs. We want to match how [va-text-input](https://design.va.gov/components/form/text-input) handles errors. The error message needs to be programmatically associated (`aria-describedby` with the input so that when the error is triggered and focus needs to be shifted to the first field in the form with an error.

[VADS mock form showing desired implementation](https://staging.va.gov/mock-simple-forms-patterns/text-input)
[VADS form error handling guidance](https://design.va.gov/components/form/#error-handling)
[Additional VADS required input handling guidance](https://design.va.gov/components/form/#what-to-do-instead)

## To Reproduce

1. Go to [autosuggest prototype](https://staging.va.gov/find-locations/) and use a screenreader
2. TAB into, then out of, the location field, without typing anything to trigger the error (Note: if #22458 is complete, you will need to click 'Search' to trigger the error)
3. See the error "Enter a zip code or a city and state in the search box" display. Note that no error is announced.

## Tasks

- [ ] Ensure the `span` holding the error message is on the page at pageload. (Seems to be injected on error in its current implementation)
- [ ] Ensure the `span` with the error message has an `id`.
- [ ] On error, set error message text content.
- [ ] On error, set `aria-describedby="<ID OF ERROR MESSAGE>"` on the input.
- [ ] On error, ensure focus is set to first field with an error with `tabindex="-1"`. In this case the location field. Note that when using VADS inputs it's set in the `va-text-input` component to wrap the entire field.

 
## AC / Expected behavior
- [ ] GIVEN a user is on facility locator prototype AND using a screenreader, WHEN an error is triggered, THEN an announcement is made to screen readers informing them of the error.
- [ ] Requires accessibility review

## Screenshots
Screenshots are from the mock form and show suggested markup.
<img width="939" height="194" alt="Code sample with aria-describedby markup." src="https://github.com/user-attachments/assets/d7b4dd42-35ff-4248-b99c-152fb218731d" />
<img width="1455" height="238" alt="Code sample showing where to put the tabindex." src="https://github.com/user-attachments/assets/ddecb10a-05fb-4033-b182-5b261bb30122" />

 ## Additional context
**Critical for AT user testing**
(Add link to research epic?)

## Definition of done
- [x] Code completed; PR created
- [ ] Review instance created/available; internal team notified to start reviews
- [ ] Code review (meets established quality and security standards)
- [ ] Design review (meets design & UX requirements)
- [ ] A11y review (meets accessibility standards)
- [ ] Functional/feature review (meets acceptance criteria)
- [ ] Automated tests passed; PR approved
- [ ] Testing completed; dev team notified to deploy
- [ ] PO/PM/external verify in prod
- [ ] Documentation updated

**Comment by carlyMhayter** [comment]
I have left a comment relevant to this issue on [this ticket.](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22458)

**Comment by qjac** [comment]
1. Does setting tabindex=-1 make the input not able to be tabbed to? As non-interactive elements, it's ok for the wrapping container or the error message itself to not be tabbable. But we do want the input to remain interactive. 
- Are you [moving focus using `focus( )` without setting tabindex](https://github.com/department-of-veterans-affairs/vets-website/compare/main...Autosuggest/22473_location_input_must_announce_errors_to_screenreaders)? 
- Is that working as expected? I'm not seeing a focus ring in the screenshot... Does it get focus? Does it display focus styling? 
- My understanding is that setting tabindex is needed to programmatically set focus on a non-interactive (and not focusable by default) element, but I can research more/ask around to confirm.

2. The id of the error message needs to match the aria-describedby on the associated input. And each input/error message in the form would need a unique id/aria-describedby pair. I'm not sure how you set a new attribute on the error message with the VA setup, but that is what needs to happen (paging @bryan-thompsoncodes, who may have some answers).

Additionally, [I've asked if there's updated guidance on error handling](https://dsva.slack.com/archives/C0335Q88X0S/p1761770205129869?thread_ts=1761232063.838689&cid=C0335Q88X0S) and I'll update when I have an answer.
[Some recent discovery on error handling](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/22458#issuecomment-3463880392)

**Comment by carlyMhayter** [comment]
Update: because central issue described in this ticket (screenreader not announcing errors on an input) was due to be addressed by VDS at some point in the future, it was decided by Facilities team that instead of addressing the issue here in this ticket, [some simple accessibility changes were to be made instead](https://github.com/department-of-veterans-affairs/vets-website/pull/39909), in order to 1) not accrue technical debt and 2) make it easier to find and address this issue when the VDS recommendations are issued.

**Comment by carlyMhayter** [comment]
Accessibility changes can be observed in [this review instance](http://aed759898e76b59c4778943782af4760.vfs.va.gov/) @qjac ([work is described in this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/39909))

**Comment by qjac** [comment]
This one needs some rework @carlyMhayter. I am seeing an aria reference error. 

<img width="1824" height="766" alt="Image" src="https://github.com/user-attachments/assets/192403b0-ad07-4a61-ba26-011c812e29f4" />

It looks like it's due to the aria-describedby added to the location input. There's no matching element with id='input-error-message' on the page. 
It also looks like the error message span has a unnecessary tabindex='-1'.

<img width="966" height="590" alt="Image" src="https://github.com/user-attachments/assets/4401684f-0fa7-4b8f-8d7d-d97b842789c5" />

- [ ] Remove the tabindex from the `span.usa-input-error-message` in the location field.
- [ ] Add a unique id to the `span.usa-input-error-message` in the location field.
- [ ] Use that same unique id as the reference for the aria-describedby on the location input.
- [ ] Focus the input on error. (note: this is updated guidance from when original AC were written, but let's do it at this point)

**Comment by r-bartlett-gsa** [comment]
@qjac Can you please give this one more a11y review.

**Comment by r-bartlett-gsa** [comment]
@qjac I asked you to review this one before realizing that this made to staging but not prod. Please review it on staging: https://staging.va.gov/find-locations/

**Comment by r-bartlett-gsa** [comment]
This is blocked because of the inability to preview the code on an RI.

**Comment by JonathanKatora** [comment]
@r-bartlett-gsa  should this be closed if it's still blocked?

**Comment by r-bartlett-gsa** [comment]
@JonathanKatora It got auto-closed, because of the PR it is linked to was merged. I don't have a strong preference on reopening it or keeping closed, as long as the status on the board is correct, and it goes through the testing process once we have RIs again.

**Comment by bryan-thompsoncodes** [comment]
@r-bartlett-gsa the latest review instance spun up successfully and is now available [here](http://0bb074dd5005aada20f79983c540368f.vfs.va.gov/)
